File: readability.c
#include<ctype.h>
#include<stdio.h>
#include<string.h>
#include<math.h>
int count_letters(char text[2000]);
int count_words(char text[2000]);
int count_sentences(char text[2000]);
int main(void)
{
    char text[2000];

 //   int words;
 //  int space;
   /* do{
        printf("Text: ");

          scanf("%s",&text);
        for(int i=0; text[i]!='\0'; i++){

            if(ispunct(text[i]) || isspace(text[i])){

                 int word=1;
                 break;
        }
       }

      }while(word!=1); */
            printf("Text: ");

            fgets(text,sizeof(text),stdin);

           // printf("%s",text);
        int letters =count_letters(text);
        int words = count_words(text);
        int sentences = count_sentences(text);

          float L = (float)((letters*100)/words);
          float S = (float)((sentences*100)/words);

          float index = (0.0588 * L) - (0.296 * S) - 15.8;
           int r_index= round(index);
        printf("%d\n",r_index);

        if(r_index<1){

            printf("Before grade 1");
            }

        else if(r_index<16)
            {

            printf("Grade %d",r_index);

            }

        else {

            printf("Grade 16+");
        }



}

int count_letters(char text[2000]){

    int num=0;

   for(int i=0; text[i]!='\0'; i++){

        if(isalpha(text[i])){

               num++;

            }

          }

   return num;

}

int count_words(char text[2000]){

    int num1=0;

    for(int i =0; text[i]!='\0';i++)
       {



        if(isspace(text[i]))
        {
          num1++;
        if(isspace(text[i+1]))
        {
            i++;
        }

         }
       }

       return num1;

}

int count_sentences(char text[2000]){

      int num2=0;
      for(int i=0; text[i]!='\0'; i++)
         {
            if(text[i]=='.' || text[i]=='?' || text[i]=='!')
              {

                   num2++;

              }


         }

       return num2;



}

File: readability.py
from cs50 import get_string


def main():

     text = get_string("Text : ")
     letters = count_letters(text)
     words = count_words(text)

     sentences = count_sentences(text)

     L = float (((letters * 100) / words))
     S = float((sentences * 100) / words)

     index = (0.0588 * L) - (0.296 * S) - 15.8
     r_index = round(index)

     if r_index<1:
          print("before grade 1")
     elif r_index <16:
          print("Grade ", r_index)
     else:
          print("Grade 16+")



def count_letters(input_string):
    count = 0
    for char in input_string:
        if char.isalpha():
            count += 1
    return count


def count_words(input_string):
    words = input_string.split()
    return len(words)




def count_sentences(str1):

    count=0

    for str in str1:
        if(str=='!' or str =="?" or str =="."):
                count+=1

    return count


main()